core/integration/services_test.go

```diff
@@ -2257,6 +2257,95 @@ server {
 	return svc, url
 }
 
+func gitSmartHTTPServiceDirAuth(ctx context.Context, t testing.TB, c *dagger.Client, hostname string, dir *dagger.Directory, username string, token *dagger.Secret) (*dagger.Service, string) {
+	t.Helper()
+
+	if username == "" {
+		username = "x-access-token"
+	}
+
+	var tokenPlain string
+	if token != nil {
+		var err error
+		tokenPlain, err = token.Plaintext(ctx)
+		require.NoError(t, err)
+	}
+
+	conf := `
+server.document-root = "/var/empty"
+server.port = 80
+server.modules = (
+  "mod_auth",
+  "mod_authn_file",
+  "mod_cgi",
+  "mod_setenv",
+  "mod_accesslog",
+  "mod_rewrite"
+)
+
+accesslog.filename = "/dev/fd/1"
+server.errorlog    = "/dev/fd/2"
+
+setenv.add-environment = (
+  "GIT_PROJECT_ROOT"    => "/var/www",
+  "GIT_HTTP_EXPORT_ALL" => ""
+)
+
+url.rewrite-if-not-file = ( "^/(.*)$" => "/git-http-backend.cgi/$1" )
+cgi.assign = ( ".cgi" => "" )
+`
+	if token != nil {
+		conf += `
+auth.backend = "plain"
+auth.backend.plain.userfile = "/etc/lighttpd/.htpasswd"
+auth.require = ( "/" => ( "method" => "basic", "realm" => "Git", "require" => "valid-user" ) )
+`
+	}
+
+	ctr := c.Container().
+		From("debian:bookworm-slim").
+		WithExec([]string{"bash", "-lc", `
+set -eux
+apt-get update
+apt-get install -y --no-install-recommends git lighttpd ca-certificates
```

Comment:
Is there a reason we're using lightttpd here over nginx? A lot of our other tests use nginx, would be good to be consistent.